packages/interaction-tracking/src/InteractionTracking.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import invariant from 'shared/invariant';
+
+import {
+  getCurrentContext as getZoneCurrentContext,
+  track as trackZone,
+  wrap as wrapZone,
+} from './InteractionZone';
+
+// TODO This package will likely want to override browser APIs (e.g. setTimeout, fetch)
+// So that async callbacks are automatically wrapped with the current tracked event info.
+// For the initial iteration, async callbacks must be explicitely wrapped with wrap().
+
+export type Interaction = {|
+  children: Array<Interaction> | null,
+  name: string,
+  timestamp: number,
+|};
+
+// Normally we would use the current renderer HostConfig's "now" method,
+// But since interaction-tracking will be a separate package,
+// I instead just copied the approach used by ReactScheduler.
+let now;
+if (typeof performance === 'object' && typeof performance.now === 'function') {
+  const localPerformance = performance;
+  now = function() {
+    return localPerformance.now();
+  };
+} else {
+  const localDate = Date;
+  now = function() {
+    return localDate.now();
+  };
+}
+
+export function track(name: string, callback: Function): void {
+  if (!__PROFILE__) {
```

Comment:
I'm missing where we allow you to finish the interaction - at some point we need to allow removing an interaction from the interactions array, right?

Reply:
Good question!

It's just a matter of JS scope. The interaction goes out of scope when the callback finishes executing. (If you've "wrapped" async callbacks, they will re-surface the interaction– since their scope is preserved as long as you have a reference to them).

Reply:
Ooooh right. I remember now, thanks.